tests/regression/diffkabi/diffkabi_test.py

```diff
@@ -0,0 +1,147 @@
+    return result
+
+
+specs = collect_task_specs()
+
+
+class TaskSpec:
+    """
+    Task specification representing one testing scenario (a kernel module with
+    a parameter)
+    """
+    def __init__(self, spec):
+        # Values from the YAML file
+        self.old_kernel = spec["old_kernel"]
+        self.new_kernel = spec["new_kernel"]
+        if "control_flow_only" in spec:
+            self.control_flow_only = spec["control_flow_only"]
+        else:
+            self.control_flow_only = False
+        self.debug = spec["debug"] if "debug" in spec else False
+        self.functions = spec["functions"]
+
+
+def prepare_task(spec):
+    """
+    Prepare testing task (scenario). Build old and new modules and copy
+    created files.
+    """
+    spec.built_functions = dict()
+
+    for function, desc in spec.functions.iteritems():
+        # Find the modules
+        first_builder = LlvmKernelBuilder(spec.old_kernel, None,
+                                          debug=spec.debug)
+        first_builder.source.build_cscope_database()
+        second_builder = LlvmKernelBuilder(spec.new_kernel, None,
+                                           debug=spec.debug)
+        second_builder.source.build_cscope_database()
+
+        # Build the modules
+        mod_first = first_builder.build_llvm_function(function)
+        mod_second = second_builder.build_llvm_function(function)
+
+        # Add the function to the dictionary
+        fun = (function, mod_first, mod_second)
+        spec.built_functions[fun] = Result[desc.upper()]
+
+        # The modules were already built when finding their sources.
+        # Now the files need only to be copied to the right place.
+        #
+        # Note: the files are copied to the task directory for reference only.
+        # The exact name of the module is not known before building the
+        # function, therefore building from module sources in kernel_modules or
+        # using already built LLVM IR files is not possible.
+        mod_path_old = mod_first.llvm
+        mod_path_new = mod_second.llvm
+        mod_old = os.path.basename(mod_first.name)
+        mod_new = os.path.basename(mod_second.name)
+        task_dir = os.path.join(tasks_path, mod_old)
+        if not os.path.isdir(tasks_path):
+            os.mkdir(tasks_path)
+        if not os.path.isdir(task_dir):
+            os.mkdir(task_dir)
+        old_src = os.path.join(task_dir, mod_old + "_old.c")
+        new_src = os.path.join(task_dir, mod_new + "_new.c")
+        old_llvm = os.path.join(task_dir, mod_old + "_old.ll")
+        new_llvm = os.path.join(task_dir, mod_new + "_new.ll")
+        old_source_src = os.path.splitext(mod_path_old)[0] + ".c"
+        new_source_src = os.path.splitext(mod_path_new)[0] + ".c"
+
+        shutil.copyfile(mod_path_old, old_llvm)
+        shutil.copyfile(old_source_src, old_src)
+
+        shutil.copyfile(mod_path_new, new_llvm)
+        shutil.copyfile(new_source_src, new_src)
```

Comment:
Couldn't we use the prepare_module function from diffkabi/regression_test? It does basically the same copying, except that it builds a module instead of a single file for a function. I'd rather have this unified somehow.